tests/modules/reply.c

```diff
@@ -0,0 +1,171 @@
+
+    return RedisModule_ReplyWithLongLong(ctx, integer);
+}
+
+int rw_double(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc != 2) return RedisModule_WrongArity(ctx);
+
+    double dbl;
+    if (RedisModule_StringToDouble(argv[1], &dbl) != REDISMODULE_OK)
+        return RedisModule_ReplyWithError(ctx, "Arg cannot be parsed as a double");
+
+    return RedisModule_ReplyWithDouble(ctx, dbl);
+}
+
+int rw_array(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc != 2) return RedisModule_WrongArity(ctx);
+
+    long long integer;
+    if (RedisModule_StringToLongLong(argv[1], &integer) != REDISMODULE_OK)
+        return RedisModule_ReplyWithError(ctx, "Arg cannot be parsed as a integer");
+
+    RedisModule_ReplyWithArray(ctx, integer);
+    for (int i = 0; i < integer; ++i) {
+        RedisModule_ReplyWithLongLong(ctx, i);
+    }
+
+    return REDISMODULE_OK;
+}
+
+int rw_map(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc != 2) return RedisModule_WrongArity(ctx);
+
+    long long integer;
+    if (RedisModule_StringToLongLong(argv[1], &integer) != REDISMODULE_OK)
+        return RedisModule_ReplyWithError(ctx, "Arg cannot be parsed as a integer");
+
+    RedisModule_ReplyWithMap(ctx, integer);
+    for (int i = 0; i < integer; ++i) {
+        RedisModule_ReplyWithLongLong(ctx, i);
+        RedisModule_ReplyWithDouble(ctx, i * 1.5);
+    }
+
+    return REDISMODULE_OK;
+}
+
+int rw_set(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc != 2) return RedisModule_WrongArity(ctx);
+
+    long long integer;
+    if (RedisModule_StringToLongLong(argv[1], &integer) != REDISMODULE_OK)
+        return RedisModule_ReplyWithError(ctx, "Arg cannot be parsed as a integer");
+
+    RedisModule_ReplyWithSet(ctx, integer);
+    for (int i = 0; i < integer; ++i) {
+        RedisModule_ReplyWithLongLong(ctx, i);
+    }
+
+    return REDISMODULE_OK;
+}
+
+int rw_attribute(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc != 2) return RedisModule_WrongArity(ctx);
+
+    long long integer;
+    if (RedisModule_StringToLongLong(argv[1], &integer) != REDISMODULE_OK)
+        return RedisModule_ReplyWithError(ctx, "Arg cannot be parsed as a integer");
+
+    RedisModule_ReplyWithAttribute(ctx, integer);
+    for (int i = 0; i < integer; ++i) {
+        RedisModule_ReplyWithLongLong(ctx, i);
+    }
+
+    return REDISMODULE_OK;
+}
+
+int rw_push(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc != 2) return RedisModule_WrongArity(ctx);
+
+    long long integer;
+    if (RedisModule_StringToLongLong(argv[1], &integer) != REDISMODULE_OK)
+        return RedisModule_ReplyWithError(ctx, "Arg cannot be parsed as a integer");
+
+    RedisModule_ReplyWithPush(ctx, integer);
+    for (int i = 0; i < integer; ++i) {
+        RedisModule_ReplyWithLongLong(ctx, i);
+    }
```

Comment:
A more realistic test would be to also reply from this command (after the push), and use a deferring client to read both replies
```suggestion
    RedisModule_ReplyWithPush(ctx, integer+1);
    RedisModule_ReplyWithCStromg(ctx, "message");
    for (int i = 0; i < integer; ++i) {
        RedisModule_ReplyWithLongLong(ctx, i);
    }
    RedisModule_ReplyWithSimpleString(ctx, "OK");
```
something like
```tcl
    r deferred 1
    r rw.push 3
    assert_equal {"message" 0 1 2} [r read]
    assert_equal {OK} [r read]
    r deferred 0
```

Reply:
Push type was removed from this PR

Reply:
maybe you forgot to (git) push the removal of the pushes 8-)